contrib/segmentio/kafka.go.v0/kafka.go

```diff
@@ -86,14 +87,51 @@ func (r *Reader) ReadMessage(ctx context.Context) (kafka.Message, error) {
 		r.prev.Finish()
 		r.prev = nil
 	}
+
 	msg, err := r.Reader.ReadMessage(ctx)
 	if err != nil {
 		return kafka.Message{}, err
 	}
-	r.prev = r.startSpan(ctx, &msg)
+
+	r.prev = r.startSpan(ctx, &msg, fmt.Sprint("Consume Topic ", msg.Topic))
+
+	return msg, nil
+}
+
+// FetchMessage polls the consumer for a message. Message will be traced.
+func (r *Reader) FetchMessage(ctx context.Context) (kafka.Message, error) {
+	if r.prev != nil {
+		r.prev.Finish()
+		r.prev = nil
+	}
+	msg, err := r.Reader.FetchMessage(ctx)
+	if err != nil {
+		return kafka.Message{}, err
+	}
+	r.prev = r.startSpan(ctx, &msg, fmt.Sprint("Consume Topic ", msg.Topic, " FetchMessage"))
 	return msg, nil
 }
 
+// CommitMessages commit the given messages. Commit will be traced using the last message as reference.
+func (r *Reader) CommitMessages(ctx context.Context, msgs ...kafka.Message) error {
+	if r.prev != nil {
+		r.prev.Finish()
+		r.prev = nil
+	}
+
+	err := r.Reader.CommitMessages(ctx, msgs...)
+	if err != nil {
+		return err
+	}
+
+	if len(msgs) != 0 {
+		msg := msgs[len(msgs)-1]
+		r.prev = r.startSpan(ctx, &msg, fmt.Sprint("Consume Topic ", msg.Topic, " CommitMessages"))
```

Comment:
Isn't this creating a span from `CommitMessages()` to the next `FetchMessage()` call? In this time nothing is consumed right?

Reply:
I don't know. 🙂 

When I did https://github.com/DataDog/dd-trace-go/pull/1283 I couldn't come up with a good solution for `CommitMessages`. Because in theory it would be ok to call `FetchMessage` multiple times and then `CommitMessage` only once.
I opted for the simplest solution: make `FetchMessage` behave exactly as `ReadMessage` -> finish the span on the next call.

We've been using it in production (with a loop similar to the example [here](https://github.com/segmentio/kafka-go#explicit-commits)) fine for several weeks now.